dask/array/backends.py

```diff
@@ -296,3 +299,157 @@ def _nannumel_sparse(x, **kwargs):
     # If all dimensions are contracted, this will just be a number, otherwise we
     # want to densify it.
     return n.todense() if hasattr(n, "todense") else n
+
+
+class ArrayBackendEntrypoint(DaskBackendEntrypoint):
+    """Dask-Array version of ``DaskBackendEntrypoint``
+
+    See Also
+    --------
+    NumpyBackendEntrypoint
+    """
+
+    def __init__(self):
+        """Register data-directed dispatch functions"""
+        raise NotImplementedError
+
+    @property
+    def RandomState(self):
+        """Return the backend-specific RandomState class
+
+        For example, the 'numpy' backend simply returns
+        ``numpy.random.RandomState``.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def ones(shape, *, dtype=None, meta=None):
+        """Create an array of ones
+
+        Returns a new array having a specified shape and filled
+        with ones.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def zeros(shape, *, dtype=None, meta=None):
+        """Create an array of zeros
+
+        Returns a new array having a specified shape and filled
+        with zeros.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def empty(shape, *, dtype=None, meta=None):
+        """Create an empty array
+
+        Returns an uninitialized array having a specified shape.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def full(shape, fill_value, *, dtype=None, meta=None):
+        """Create a uniformly filled array
+
+        Returns a new array having a specified shape and filled
+        with fill_value.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def arange(start, /, stop=None, step=1, *, dtype=None, meta=None):
+        """Create an ascending or descending array
+
+        Returns evenly spaced values within the half-open interval
+        ``[start, stop)`` as a one-dimensional array.
+        """
+        raise NotImplementedError
+
+
+class NumpyBackendEntrypoint(ArrayBackendEntrypoint):
+    def __init__(self):
+        # Importing this class will already guarentee
+        # that data-dispatch functions are registered
+        pass
+
+    @property
+    def RandomState(self):
+        return np.random.RandomState
```

Comment:
It's worth noting that the NumPy API is changing here (as are other libraries like CuPy). There was an attempt to port these changes to Dask in PR ( https://github.com/dask/dask/pull/9038 ). While it is likely out-of-scope to adopt the new API here, it would be good if we don't get too anchored to the old API (and save room for upgrading in the future)

Reply:
Gotcha. Well it was mentioned in the 2022.10.0 release issue ( https://github.com/dask/community/issues/281 ) that we want to get this in for the upcoming release. So this may involve proposing changes to that PR to fix conflicts.